src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,645 @@
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+}
+
+void PushAvStreamTransportServer::HandleModifyPushTransport(HandlerContext & ctx,
+                                                            const Commands::ModifyPushTransport::DecodableType & commandData)
+{
+    Status status           = Status::Success;
+    uint16_t connectionID   = commandData.connectionID;
+    auto & transportOptions = commandData.transportOptions;
+
+    TransportConfigurationStorageWithFabricIndex * transportConfiguration = FindStreamTransportConnection(connectionID);
+
+    if (transportConfiguration == nullptr)
+    {
+        ChipLogError(Zcl, "HandleModifyPushTransport: ConnectionID Not Found.");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return;
+    }
+
+    if (transportConfiguration->fabricIndex != ctx.mCommandHandler.GetAccessingFabricIndex())
+    {
+        ChipLogError(Zcl, "HandleModifyPushTransport: ConnectionID Not Found.");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return;
+    }
+
+    if (mDelegate.GetTransportStatus(connectionID) == PushAvStreamTransportStatusEnum::kBusy)
+    {
+        ChipLogError(Zcl, "HandleModifyPushTransport: Connection is Busy");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Busy);
+        return;
+    }
+
+    TransportOptionsStruct transportOptionArgs;
+    transportOptionArgs.streamUsage                = transportOptions.streamUsage;
+    transportOptionArgs.videoStreamID              = transportOptions.videoStreamID;
+    transportOptionArgs.audioStreamID              = transportOptions.audioStreamID;
+    transportOptionArgs.endpointID                 = transportOptions.endpointID;
+    transportOptionArgs.url                        = transportOptions.url;
+    transportOptionArgs.triggerOptions.triggerType = transportOptions.triggerOptions.triggerType;
+    // Todo: copy motion zones
+    transportOptionArgs.triggerOptions.motionSensitivity = transportOptions.triggerOptions.motionSensitivity;
+    transportOptionArgs.triggerOptions.motionTimeControl = transportOptions.triggerOptions.motionTimeControl;
+    transportOptionArgs.triggerOptions.maxPreRollLen     = transportOptions.triggerOptions.maxPreRollLen;
+    transportOptionArgs.ingestMethod                     = transportOptions.ingestMethod;
+    transportOptionArgs.containerOptions                 = transportOptions.containerOptions;
+    transportOptionArgs.expiryTime                       = transportOptions.expiryTime;
+
+    // Call the delegate
+    status = mDelegate.ModifyPushTransport(connectionID, transportOptionArgs);
+
+    if (status == Status::Success)
+    {
+        if (transportConfiguration->transportConfiguration.transportOptions.HasValue())
+        {
+            transportConfiguration->transportConfiguration.transportOptions =
+                static_cast<Optional<Structs::TransportOptionsStruct::Type>>(transportOptionArgs);
```

Comment:
MakeOptional, if that's what we are after here.

But this just looks broken; it's stashing pointers to the network buffer into the transportConfiguration, so this is use-after-free.